### PR TITLE
[quant] Adding Scalar add/mul.

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25080 [quant] Rewriting the scalar add
* #24873 [quant] Rewrite of the scalar mul
* **#24447 [quant] Adding Scalar add/mul.**

Note: This should be landed ONLY after #24259

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24447

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)

### CODE REVIEW

**Review by lly-zero-one — COMMENTED** [review]
LGTM, can you also add a test to cover the add/mul scalar?

File: aten/src/ATen/native/quantized/cpu/qadd.cpp [hunk 1]
```diff
@@ -48,6 +48,28 @@ Tensor _add_out(Tensor& out, const Tensor& self, const Tensor& other) {
...
+  int64_t zero_point = out.q_zero_point();
+  double scale = out.q_scale();
+  int64_t self_zero_point = self.q_zero_point();
+  double self_scale = self.q_scale();
+
+  auto iter = TensorIterator::unary_op(out, self);
+  AT_DISPATCH_QINT_TYPES(out.scalar_type(), "qadd", [&]() {
+    cpu_kernel(iter, [&](scalar_t a) -> scalar_t {
+      const auto da = at::dequantize_val(self_scale, self_zero_point, a);
+      float c = da + float(other);
```
Comment (llyfacebook) [inline aten/src/ATen/native/quantized/cpu/qadd.cpp]: call the scalar to float conversion function?

**Comment by zafartahirov** [comment]
> LGTM, can you also add a test to cover the add/mul scalar?

The numerical tests are performed on the kernels. The API test is covered by the scalability

File: aten/src/ATen/native/quantized/cpu/qadd.cpp [hunk 2]
```diff
@@ -66,30 +88,69 @@ class QAdd final : public c10::OperatorKernel {
 template <bool ReLUFused = false>
 class QAddOut final : public c10::OperatorKernel {
  public:
-  Tensor operator()(at::Tensor qa, at::Tensor qb, at::Tensor out) {
+  Tensor operator()(Tensor qa, Tensor qb, at::Tensor out) {
```
Comment (llyfacebook) [inline aten/src/ATen/native/quantized/cpu/qadd.cpp]: Can you make the namespace be consistent?

**Review by lly-zero-one — APPROVED** [review]
looks good!

**Pull request closed without merging**